lib/daru/maths/statistics/vector.rb

```diff
@@ -377,6 +377,39 @@ def sample_without_replacement(sample=1)
           end
         end
 
+        # The percent_change method computes the percent change over
+        # the given number of periods.
+        #
+        # @param [Integer] periods (1) number of nils to insert at the beginning.
+        #
+        # @example
+        # 
+        #   vector = Daru::Vector.new([4,6,6,8,10],index: ['a','f','t','i','k'])
+        #   vector.percent_change
+        #   #=>
+        #   #   <Daru::Vector:28713060 @name = nil @size: 5 >
+        #   #              nil
+        #   #   a	
+        #   #   f	   0.5
+        #   #   t	   0.0
+        #   #   i	   0.3333333333333333
+        #   #   k          0.25
+        def percent_change periods = 1
+          type == :numeric or raise TypeError, "Vector must be numeric"
+          arr = []
+          i = periods
+          (periods...size).each do |j|
+            if @missing_values.has_key?(@data[i])
+              arr[j] = nil
+              @data.delete_at(i)
```

Comment:
Didn't you notice that `delete_at` changes the data of the vector on which percent_change is being called?